src/domains.ts

```diff
@@ -0,0 +1,325 @@
+        const errorRes = await response.json()
+        throw this.domainError.HandleError(errorRes.error, response.status)
+      }
+      const domainData = await response.json()
+      return domainData
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  public async ListUserDomains (userID: string, queryParams: QueryParams, token: string): Promise<DomainsInterface> {
+    // ListUserDomains retrieves all domains for a user.
+
+    const stringParams: Record<string, string> = Object.fromEntries(
+      Object.entries(queryParams).map(([key, value]) => [key, String(value)])
+    )
+    const options: RequestInit = {
+      method: 'GET',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    try {
+      const response = await fetch(
+        new URL(`/users/${userID}/domains?${new URLSearchParams(stringParams).toString()}`, this.domainsUrl).toString(),
+        options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.domainError.HandleError(errorRes.error, response.status)
+      }
+      const domainData = await response.json()
+      return domainData
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  public async EnableDomain (domainID: string, token: string): Promise<Domain> {
+    // EnableDomain enables domain with provided ID.
+    const options: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    try {
+      const response = await fetch(
+        new URL(`${this.domainsEndpoint}/${domainID}/enable`, this.domainsUrl).toString(),
+        options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.domainError.HandleError(errorRes.error, response.status)
+      }
+      const domainData: Domain = await response.json()
+      return domainData
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  public async DisableDomain (domainID: string, token: string): Promise<Domain> {
+    // DisableDomain disables domain with provided ID.
+    const options: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    try {
+      const response = await fetch(
+        new URL(`${this.domainsEndpoint}/${domainID}/disable`, this.domainsUrl).toString(),
+        options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.domainError.HandleError(errorRes.error, response.status)
+      }
+      const domainData: Domain = await response.json()
+      return domainData
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  public async AddUsertoDomain (domainID: string, userIDs: string[], relation: string, token: string): Promise<string> {
```

Comment:
same for this. return `statusCode`, or you can create a return interface, that will return the status code and the message.